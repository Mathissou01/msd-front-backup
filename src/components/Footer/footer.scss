@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/01-settings/settings.layout" as layout;
@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/02-tools/tools.include-media" as im;

.c-Footer {
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-top: 1px solid colors.$gray-200;

  &__ContentContainer {
    $self: &;

    display: flex;
    flex-wrap: wrap;
    row-gap: 8px;
    align-items: center;
    justify-content: center;
    width: 92%;
    padding: 16px 12px;

    @include im.media(">=mobile") {
      min-height: 69px;
    }
  }

  &__Link {
    @include typo.text("footer");

    position: relative;
    display: flex;
    align-items: center;
    height: 21px;
    padding-right: 12px;
    padding-left: 12px;
    transition: text-decoration 0.3s ease-in-out;

    @include im.media(">=mobile") {
      padding: 0 20px;
    }

    :hover {
      text-decoration: underline;
    }

    &::before {
      position: absolute;
      right: 0;
      width: 1px;
      height: 20px;
      background: colors.$gray-500;
      content: "";
    }

    &:last-child {
      &::before {
        content: none;
      }
    }
  }

  &__Version {
    margin-right: 1em;
    font-size: 16px;
  }
}
