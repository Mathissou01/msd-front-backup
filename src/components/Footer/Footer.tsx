import Link from "next/link";
import { useContract } from "../../hooks/useContract";
import "./footer.scss";

enum EAccessibilityLevel {
  "not_conform" = "non conforme",
  "partially_conform" = "partiellement conforme",
  "conform" = "totalement conforme",
}

export default function Footer() {
  /* Static Data */
  const labels = {
    accessibilityLabel: "Accessibilité : ",
    siteLabel: "Le site de SUEZ",
    cguLabel: "Conditions générales",
    cookiesLabel: "Politique de cookies",
    confidentialityLabel: "Politique de confidentialité",
    contactUsSubService: "Contactez-nous",
  };
  const routes = {
    accessibilityRoute: "/accessibilite",
    siteRoute: "/",
    cguRoute: "/",
    cookiesRoute: "/",
    confidentialityRoute: "/",
    contactUsRoute: "/service/contact",
  };

  /* Global Data */
  const { contract } = useContract();
  const footerData =
    contract?.attributes?.contractCustomization?.data?.attributes?.footer?.data
      ?.attributes;

  const accessibilityLevelLabel = footerData?.accessibilityLevel
    ? Object.values(EAccessibilityLevel)[
        Object.keys(EAccessibilityLevel).indexOf(footerData.accessibilityLevel)
      ]
    : EAccessibilityLevel.not_conform;

  return (
    <footer className="c-Footer" data-testid="footer">
      <Link className="c-Footer__Link" href={routes.accessibilityRoute}>
        <span>{labels.accessibilityLabel + accessibilityLevelLabel}</span>
      </Link>
      <Link className="c-Footer__Link" href={routes.siteRoute}>
        <span>{labels.siteLabel}</span>
      </Link>
      <Link
        className="c-Footer__Link"
        href={
          footerData?.cguSubService?.data?.attributes?.link ?? routes.cguRoute
        }
      >
        <span>{labels.cguLabel}</span>
      </Link>
      <Link
        className="c-Footer__Link"
        href={
          footerData?.cookiesSubService?.data?.attributes?.link ??
          routes.cookiesRoute
        }
      >
        <span>{labels.cookiesLabel}</span>
      </Link>
      <Link
        className="c-Footer__Link"
        href={
          footerData?.confidentialitySubService?.data?.attributes?.link ??
          routes.confidentialityRoute
        }
      >
        <span>{labels.confidentialityLabel}</span>
      </Link>

      {footerData?.contactUsSubService?.data?.attributes?.isActivated &&
        footerData?.contactUsSubService?.data.attributes.label && (
          <Link
            className="c-Footer__Link"
            href={
              footerData.contactUsSubService?.data?.attributes.link ??
              routes.contactUsRoute
            }
          >
            <span>{footerData.contactUsSubService.data?.attributes.label}</span>
          </Link>
        )}
    </footer>
  );
}
