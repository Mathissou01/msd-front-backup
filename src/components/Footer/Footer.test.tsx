import { render, screen } from "@testing-library/react";
import Footer from "./Footer";

describe("Header", () => {
  it("renders", async () => {
    const { container } = render(<Footer />);

    const footer = screen.getByTestId("footer");
    expect(footer).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
