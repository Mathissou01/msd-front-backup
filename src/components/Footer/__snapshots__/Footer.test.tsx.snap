// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div>
  <footer
    class="c-Footer"
    data-testid="footer"
  >
    <a
      class="c-Footer__Link"
      href="/accessibilite"
    >
      <span>
        Accessibilité : non conforme
      </span>
    </a>
    <a
      class="c-Footer__Link"
      href="/"
    >
      <span>
        Le site de SUEZ
      </span>
    </a>
    <a
      class="c-Footer__Link"
      href="/"
    >
      <span>
        Conditions générales
      </span>
    </a>
    <a
      class="c-Footer__Link"
      href="/"
    >
      <span>
        Politique de cookies
      </span>
    </a>
    <a
      class="c-Footer__Link"
      href="/"
    >
      <span>
        Politique de confidentialité
      </span>
    </a>
  </footer>
</div>
`;
